--- gst_methods.defs	2009-05-18 19:28:48.000000000 -0400
+++ gst_methods.defs.new	2009-05-18 19:28:15.000000000 -0400
@@ -1,4 +1,7 @@
 ;; -*- scheme -*-
+
+;; Apply gst_methods.defs.patch after regenerating.
+
 ; object definitions ...
 (define-object Adapter
   (in-module "Gst")
@@ -540,46 +543,6 @@
   )
 )
 
-(define-enum EventType
-  (in-module "Gst")
-  (c-name "GstEventType")
-  (gtype-id "GST_TYPE_EVENT_TYPE")
-  (values
-    '("gst-event-unknown" "GST_EVENT_UNKNOWN")
-    '("0)" "0)")
-    '("gst-event-flush-start" "GST_EVENT_FLUSH_START")
-    '("flag(both))" "FLAG(BOTH))")
-    '("gst-event-flush-stop" "GST_EVENT_FLUSH_STOP")
-    '("flag(both)" "FLAG(BOTH)")
-    '("gst-event-eos" "GST_EVENT_EOS")
-    '("flag(downstream)" "FLAG(DOWNSTREAM)")
-    '("gst-event-newsegment" "GST_EVENT_NEWSEGMENT")
-    '("flag(downstream)" "FLAG(DOWNSTREAM)")
-    '("gst-event-tag" "GST_EVENT_TAG")
-    '("flag(downstream)" "FLAG(DOWNSTREAM)")
-    '("gst-event-buffersize" "GST_EVENT_BUFFERSIZE")
-    '("flag(downstream)" "FLAG(DOWNSTREAM)")
-    '("gst-event-qos" "GST_EVENT_QOS")
-    '("flag(upstream))" "FLAG(UPSTREAM))")
-    '("gst-event-seek" "GST_EVENT_SEEK")
-    '("flag(upstream))" "FLAG(UPSTREAM))")
-    '("gst-event-navigation" "GST_EVENT_NAVIGATION")
-    '("flag(upstream))" "FLAG(UPSTREAM))")
-    '("gst-event-latency" "GST_EVENT_LATENCY")
-    '("flag(upstream))" "FLAG(UPSTREAM))")
-    '("gst-event-custom-upstream" "GST_EVENT_CUSTOM_UPSTREAM")
-    '("flag(upstream))" "FLAG(UPSTREAM))")
-    '("gst-event-custom-downstream" "GST_EVENT_CUSTOM_DOWNSTREAM")
-    '("flag(downstream)" "FLAG(DOWNSTREAM)")
-    '("gst-event-custom-downstream-oob" "GST_EVENT_CUSTOM_DOWNSTREAM_OOB")
-    '("flag(downstream))" "FLAG(DOWNSTREAM))")
-    '("gst-event-custom-both" "GST_EVENT_CUSTOM_BOTH")
-    '("flag(both)" "FLAG(BOTH)")
-    '("gst-event-custom-both-oob" "GST_EVENT_CUSTOM_BOTH_OOB")
-    '("flag(both))" "FLAG(BOTH))")
-  )
-)
-
 (define-enum SeekType
   (in-module "Gst")
   (c-name "GstSeekType")
@@ -4285,6 +4248,13 @@
   (return-type "GQuark")
 )
 
+;; This method is not generated for some reason.
+(define-method get_seqnum
+  (of-object "GstMessage")
+  (c-name "gst_message_get_seqnum")
+  (return-type "guint32")
+)
+
 (define-method set_seqnum
   (of-object "GstMessage")
   (c-name "gst_message_set_seqnum")
