diff --git a/gstreamer/src/gst_enums.defs b/gstreamer/src/gst_enums.defs
index 41a303f..faf46c8 100644
--- a/gstreamer/src/gst_enums.defs
+++ b/gstreamer/src/gst_enums.defs
@@ -2891,10 +2891,11 @@
 ;;   GST_QUEUE_LEAK_DOWNSTREAM     = 2
 ;; };
 
-(define-enum-extended 
+(define-enum-extended QueueLeaky
   (in-module "Gst")
-  (c-name "")
+  (c-name "GstQueueLeaky")
   (values
+    '("no" "GST_QUEUE_LEAK_NO_LEAK" "1")
     '("upstream" "GST_QUEUE_LEAK_UPSTREAM" "1")
     '("downstream" "GST_QUEUE_LEAK_DOWNSTREAM" "2")
   )
@@ -6768,9 +6769,9 @@
   (in-module "Gst")
   (c-name "GstBaseTextOverlayLineAlign")
   (values
-    '("left" "GST_BASE_TEXT_OVERLAY_LINE_ALIGN_LEFT" "PANGO_ALIGN_LEFT")
-    '("center" "GST_BASE_TEXT_OVERLAY_LINE_ALIGN_CENTER" "PANGO_ALIGN_CENTER")
-    '("right" "GST_BASE_TEXT_OVERLAY_LINE_ALIGN_RIGHT" "PANGO_ALIGN_RIGHT")
+    '("left" "GST_TEXT_RENDER_LINE_ALIGN_LEFT" "0")
+    '("center" "GST_TEXT_RENDER_LINE_ALIGN_CENTER" "1")
+    '("right" "GST_TEXT_RENDER_LINE_ALIGN_RIGHT" "2")
   )
 )
 
@@ -6821,9 +6822,9 @@
   (in-module "Gst")
   (c-name "GstTextRenderLineAlign")
   (values
-    '("left" "GST_TEXT_RENDER_LINE_ALIGN_LEFT" "PANGO_ALIGN_LEFT")
-    '("center" "GST_TEXT_RENDER_LINE_ALIGN_CENTER" "PANGO_ALIGN_CENTER")
-    '("right" "GST_TEXT_RENDER_LINE_ALIGN_RIGHT" "PANGO_ALIGN_RIGHT")
+    '("left" "GST_TEXT_RENDER_LINE_ALIGN_LEFT" "0")
+    '("center" "GST_TEXT_RENDER_LINE_ALIGN_CENTER" "1")
+    '("right" "GST_TEXT_RENDER_LINE_ALIGN_RIGHT" "2")
   )
 )
 
@@ -6866,3 +6867,70 @@
   )
 )
 
+;; Handwritten for CdParanoiaSrc plugin (values taken from output of
+;; `gst-inspect cdparanoiasrc' - "paranoia-mode" property information)
+(define-enum-extended CdParanoiaMode
+  (in-module "Gst")
+  (c-name "GstCdParanoiaMode")
+  (values
+    '("disable" "GST_PARANOIA_MODE_DISABLE" "1<<0")
+    '("fragment" "GST_PARANOIA_MODE_FRAGMENT" "1<<1")
+    '("overlap" "GST_PARANOIA_MODE_OVERLAP" "1<<2")
+    '("scratch" "GST_PARANOIA_MODE_SCRATCH" "1<<3")
+    '("repair" "GST_PARANOIA_MODE_REPAIR" "1<<4")
+    '("full" "GST_PARANOIA_MODE_FULL" "0xff")
+  )
+)
+
+;; Handwritten for EncodeBin plugin (values taken from file
+;; gst-plugins-base/gst/encoding/gstencodebin.c, enum used for 'flags' property)
+(define-enum-extended EncodeBinFlags
+  (in-module "Gst")
+  (c-name "GstEncodeBinFlags")
+  (values
+    '("no-audio-conversion" "GST_ENCODEBIN_FLAG_NO_AUDIO_CONVERSION" "1<<0")
+    '("no-video-conversion" "GST_ENCODEBIN_FLAG_NO_VIDEO_CONVERSION" "1<<1")
+  )
+)
+
+;; Handwritten for MultiSocketSink plugin (values taken from output of
+;; `gst-inspect multisocketsink' - "sync-method" property information)
+(define-enum-extended MultiHandleSinkSyncMethod
+  (in-module "Gst")
+  (c-name "GstMultiHandleSinkSyncMethod")
+  (values
+    '("latest" "GST_SYNC_METHOD_LATEST" "0")
+    '("next-keyframe" "GST_SYNC_METHOD_NEXT_KEYFRAME" "1")
+    '("latest-keyframe" "GST_SYNC_METHOD_LATEST_KEYFRAME" "2")
+    '("burst" "GST_SYNC_METHOD_BURST" "3")
+    '("burst-keyframe" "GST_SYNC_METHOD_BURST_KEYFRAME" "4")
+    '("burst-with-keyframe" "GST_SYNC_METHOD_BURST_WITH_KEYFRAME" "5")
+  )
+)
+
+;; Handwritten for MultiSocketSink plugin (values taken from file
+;; gst-plugins-base/gst/tcp/gstmultihandlesink.c, enum used for 'client-removed' signal)
+(define-enum-extended MultiHandleSinkClientStatus
+  (in-module "Gst")
+  (c-name "GstMultiHandleSinkClientStatus")
+  (values
+    '("ok" "GST_CLIENT_STATUS_OK" "0")
+    '("closed" "GST_CLIENT_STATUS_CLOSED" "1")
+    '("removed" "GST_CLIENT_STATUS_REMOVED" "2")
+    '("slow" "GST_CLIENT_STATUS_SLOW" "3")
+    '("error" "GST_CLIENT_STATUS_ERROR" "4")
+    '("duplicate" "GST_CLIENT_STATUS_DUPLICATE" "5")
+    '("flushing" "GST_CLIENT_STATUS_FLUSHING" "6")
+  )
+)
+
+;; Handwritten for PlaySink plugin (values taken from file
+;; gst-plugins-base/gst/playback/gstplaysink.c, enum used for 'send-event-mode' property)
+(define-enum-extended PlaySinkSendEventMode
+  (in-module "Gst")
+  (c-name "GstPlaySinkSendEventMode")
+  (values
+    '("default" "MODE_DEFAULT" "0")
+    '("first" "MODE_FIRST" "1")
+  )
+)
