--- gst_methods.defs.orig	2010-03-15 16:29:33.000000000 -0400
+++ gst_methods.defs	2010-03-15 16:33:32.000000000 -0400
@@ -1,11 +1,14 @@
 ;; -*- scheme -*-
+
+;; Apply gst_methods.defs.patch after regenerating.
+
 ; object definitions ...
 (define-object Adapter
   (in-module "Gst")
   (parent "GObject")
   (c-name "GstAdapter")
   (gtype-id "GST_TYPE_ADAPTER")
 )
 
 (define-object Buffer
   (in-module "Gst")
@@ -272,21 +275,21 @@
   (gtype-id "GST_TYPE_XML")
 )
 
 ;; Enumerations and flags ...
 
 (define-flags BinFlags
   (in-module "Gst")
   (c-name "GstBinFlags")
   (gtype-id "GST_TYPE_BIN_FLAGS")
   (values
-    '("t" "GST_BIN_FLAG_LAST")
+    '("last" "GST_BIN_FLAG_LAST")
   )
 )
 
 (define-flags BufferFlag
   (in-module "Gst")
   (c-name "GstBufferFlag")
   (gtype-id "GST_TYPE_BUFFER_FLAG")
   (values
     '("readonly" "GST_BUFFER_FLAG_READONLY")
     '("preroll" "GST_BUFFER_FLAG_PREROLL")
@@ -321,42 +324,42 @@
     '("skip-group" "GST_BUFFER_LIST_SKIP_GROUP")
     '("end" "GST_BUFFER_LIST_END")
   )
 )
 
 (define-flags BusFlags
   (in-module "Gst")
   (c-name "GstBusFlags")
   (gtype-id "GST_TYPE_BUS_FLAGS")
   (values
-    '("ushing" "GST_BUS_FLUSHING")
-    '("ag-last" "GST_BUS_FLAG_LAST")
+    '("flushing" "GST_BUS_FLUSHING")
+    '("flag-last" "GST_BUS_FLAG_LAST")
   )
 )
 
 (define-enum BusSyncReply
   (in-module "Gst")
   (c-name "GstBusSyncReply")
   (gtype-id "GST_TYPE_BUS_SYNC_REPLY")
   (values
     '("drop" "GST_BUS_DROP")
     '("pass" "GST_BUS_PASS")
     '("async" "GST_BUS_ASYNC")
   )
 )
 
 (define-flags CapsFlags
   (in-module "Gst")
   (c-name "GstCapsFlags")
   (gtype-id "GST_TYPE_CAPS_FLAGS")
   (values
-    '("y" "GST_CAPS_FLAGS_ANY")
+    '("any" "GST_CAPS_FLAGS_ANY")
   )
 )
 
 (define-enum ClockReturn
   (in-module "Gst")
   (c-name "GstClockReturn")
   (gtype-id "GST_TYPE_CLOCK_RETURN")
   (values
     '("ok" "GST_CLOCK_OK")
     '("early" "GST_CLOCK_EARLY")
@@ -544,64 +547,20 @@
   (in-module "Gst")
   (c-name "GstEventTypeFlags")
   (gtype-id "GST_TYPE_EVENT_TYPE_FLAGS")
   (values
     '("upstream" "GST_EVENT_TYPE_UPSTREAM")
     '("downstream" "GST_EVENT_TYPE_DOWNSTREAM")
     '("serialized" "GST_EVENT_TYPE_SERIALIZED")
   )
 )
 
-(define-enum EventType
-  (in-module "Gst")
-  (c-name "GstEventType")
-  (gtype-id "GST_TYPE_EVENT_TYPE")
-  (values
-    '("gst-event-unknown" "GST_EVENT_UNKNOWN")
-    '("0)" "0)")
-    '("gst-event-flush-start" "GST_EVENT_FLUSH_START")
-    '("flag(both))" "FLAG(BOTH))")
-    '("gst-event-flush-stop" "GST_EVENT_FLUSH_STOP")
-    '("flag(both)" "FLAG(BOTH)")
-    '("gst-event-eos" "GST_EVENT_EOS")
-    '("flag(downstream)" "FLAG(DOWNSTREAM)")
-    '("gst-event-newsegment" "GST_EVENT_NEWSEGMENT")
-    '("flag(downstream)" "FLAG(DOWNSTREAM)")
-    '("gst-event-tag" "GST_EVENT_TAG")
-    '("flag(downstream)" "FLAG(DOWNSTREAM)")
-    '("gst-event-buffersize" "GST_EVENT_BUFFERSIZE")
-    '("flag(downstream)" "FLAG(DOWNSTREAM)")
-    '("gst-event-sink-message" "GST_EVENT_SINK_MESSAGE")
-    '("flag(downstream)" "FLAG(DOWNSTREAM)")
-    '("gst-event-qos" "GST_EVENT_QOS")
-    '("flag(upstream))" "FLAG(UPSTREAM))")
-    '("gst-event-seek" "GST_EVENT_SEEK")
-    '("flag(upstream))" "FLAG(UPSTREAM))")
-    '("gst-event-navigation" "GST_EVENT_NAVIGATION")
-    '("flag(upstream))" "FLAG(UPSTREAM))")
-    '("gst-event-latency" "GST_EVENT_LATENCY")
-    '("flag(upstream))" "FLAG(UPSTREAM))")
-    '("gst-event-step" "GST_EVENT_STEP")
-    '("flag(upstream))" "FLAG(UPSTREAM))")
-    '("gst-event-custom-upstream" "GST_EVENT_CUSTOM_UPSTREAM")
-    '("flag(upstream))" "FLAG(UPSTREAM))")
-    '("gst-event-custom-downstream" "GST_EVENT_CUSTOM_DOWNSTREAM")
-    '("flag(downstream)" "FLAG(DOWNSTREAM)")
-    '("gst-event-custom-downstream-oob" "GST_EVENT_CUSTOM_DOWNSTREAM_OOB")
-    '("flag(downstream))" "FLAG(DOWNSTREAM))")
-    '("gst-event-custom-both" "GST_EVENT_CUSTOM_BOTH")
-    '("flag(both)" "FLAG(BOTH)")
-    '("gst-event-custom-both-oob" "GST_EVENT_CUSTOM_BOTH_OOB")
-    '("flag(both))" "FLAG(BOTH))")
-  )
-)
-
 (define-enum SeekType
   (in-module "Gst")
   (c-name "GstSeekType")
   (gtype-id "GST_TYPE_SEEK_TYPE")
   (values
     '("none" "GST_SEEK_TYPE_NONE")
     '("cur" "GST_SEEK_TYPE_CUR")
     '("set" "GST_SEEK_TYPE_SET")
     '("end" "GST_SEEK_TYPE_END")
   )
@@ -927,22 +886,22 @@
     '("sometimes" "GST_PAD_SOMETIMES")
     '("request" "GST_PAD_REQUEST")
   )
 )
 
 (define-flags PadTemplateFlags
   (in-module "Gst")
   (c-name "GstPadTemplateFlags")
   (gtype-id "GST_TYPE_PAD_TEMPLATE_FLAGS")
   (values
-    '("ixed" "GST_PAD_TEMPLATE_FIXED")
-    '("lag-last" "GST_PAD_TEMPLATE_FLAG_LAST")
+    '("fixed" "GST_PAD_TEMPLATE_FIXED")
+    '("flag-last" "GST_PAD_TEMPLATE_FLAG_LAST")
   )
 )
 
 (define-enum ParseError
   (in-module "Gst")
   (c-name "GstParseError")
   (gtype-id "GST_TYPE_PARSE_ERROR")
   (values
     '("syntax" "GST_PARSE_ERROR_SYNTAX")
     '("no-such-element" "GST_PARSE_ERROR_NO_SUCH_ELEMENT")
@@ -4389,20 +4348,27 @@
   (c-name "gst_message_type_get_name")
   (return-type "const-gchar*")
 )
 
 (define-method to_quark
   (of-object "GstMessageType")
   (c-name "gst_message_type_to_quark")
   (return-type "GQuark")
 )
 
+;; This method is not generated for some reason.
+(define-method get_seqnum
+  (of-object "GstMessage")
+  (c-name "gst_message_get_seqnum")
+  (return-type "guint32")
+)
+
 (define-method set_seqnum
   (of-object "GstMessage")
   (c-name "gst_message_set_seqnum")
   (return-type "none")
   (parameters
     '("guint32" "seqnum")
   )
 )
 
 (define-function gst_message_new_eos
