diff --git a/gstreamer/src/gst_signals.defs b/gstreamer/src/gst_signals.defs
index 9c84fe5..a0103e2 100644
--- a/gstreamer/src/gst_signals.defs
+++ b/gstreamer/src/gst_signals.defs
@@ -179,7 +179,7 @@
   (when "first")
   (parameters
     '("GstObject*" "p0")
-    '("GParam" "p1")
+    '("GParamSpec*" "p1")
   )
 )
 
@@ -927,7 +927,7 @@
   (when "last")
   (parameters
     '("GstDiscovererInfo*" "p0")
-    '("GError*" "p1")
+    '("const-GError*" "p1")
   )
 )
 
