--- gst_signals.defs.gen	2008-10-15 17:01:33.000000000 -0400
+++ gst_signals.defs	2008-10-15 17:30:32.000000000 -0400
@@ -1,3 +1,5 @@
+;; Apply gst_signals.defs.patch after regenerating.
+
 ;; From GstBus
 
 (define-signal sync-message
@@ -5,7 +7,7 @@
   (return-type "void")
   (when "last")
   (parameters
-    '("GstMessage" "p0")
+    '("GstMessage*" "p0")
   )
 )
 
@@ -14,7 +16,7 @@
   (return-type "void")
   (when "last")
   (parameters
-    '("GstMessage" "p0")
+    '("GstMessage*" "p0")
   )
 )
 
@@ -76,7 +78,7 @@
   (return-type "void")
   (when "first")
   (parameters
-    '("GObject*" "p0")
+    '("GstObject*" "p0")
   )
 )
 
@@ -85,7 +87,7 @@
   (return-type "void")
   (when "first")
   (parameters
-    '("GObject*" "p0")
+    '("GstObject*" "p0")
   )
 )
 
@@ -351,7 +353,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GstMiniObject" "p0")
+    '("GstMiniObject*" "p0")
   )
 )
 
@@ -516,7 +518,7 @@
   (return-type "void")
   (when "last")
   (parameters
-    '("gpointer" "p0")
+    '("GstPlugin*" "p0")
   )
 )
 
@@ -525,7 +527,7 @@
   (return-type "void")
   (when "last")
   (parameters
-    '("gpointer" "p0")
+    '("GstPluginFeature*" "p0")
   )
 )
 
@@ -636,7 +638,7 @@
   (when "last")
   (parameters
     '("GstObject*" "p0")
-    '("gpointer" "p1")
+    '("xmlNode*" "p1")
   )
 )
 
