--- gstbase_enums.defs	2008-12-16 23:00:53.000000000 -0500
+++ gstbase_enums.defs.new	2008-12-16 22:50:03.000000000 -0500
@@ -707,10 +707,10 @@
   (in-module "Gst")
   (c-name "GstAudioConvertDithering")
   (values
-    '("none" "DITHER_NONE" "0")
-    '("rpdf" "DITHER_RPDF" "1")
-    '("tpdf" "DITHER_TPDF" "2")
-    '("tpdf-hf" "DITHER_TPDF_HF" "3")
+    '("none" "GST_AUDIO_CONVERT_DITHER_NONE" "0")
+    '("rpdf" "GST_AUDIO_CONVERT_DITHER_RPDF" "1")
+    '("tpdf" "GST_AUDIO_CONVERT_DITHER_TPDF" "2")
+    '("tpdf-hf" "GST_AUDIO_CONVERT_DITHER_TPDF_HF" "3")
   )
 )
 
@@ -718,24 +718,11 @@
   (in-module "Gst")
   (c-name "GstAudioConvertNoiseShaping")
   (values
-    '("none" "NOISE_SHAPING_NONE" "0")
-    '("error-feedback" "NOISE_SHAPING_ERROR_FEEDBACK" "1")
-    '("simple" "NOISE_SHAPING_SIMPLE" "2")
-    '("medium" "NOISE_SHAPING_MEDIUM" "3")
-    '("high" "NOISE_SHAPING_HIGH" "4")
-  )
-)
-
-;; From /usr/src/jhbuild-sources/gst-plugins-base/gst/audioresample/resample.h
-
-(define-enum-extended Format
-  (in-module "Resample")
-  (c-name "ResampleFormat")
-  (values
-    '("s16" "RESAMPLE_FORMAT_S16" "0")
-    '("s32" "RESAMPLE_FORMAT_S32" "1")
-    '("f32" "RESAMPLE_FORMAT_F32" "2")
-    '("f64" "RESAMPLE_FORMAT_F64" "3")
+    '("none" "GST_AUDIO_CONVERT_NOISE_SHAPING_NONE" "0")
+    '("error-feedback" "GST_AUDIO_CONVERT_NOISE_SHAPING_ERROR_FEEDBACK" "1")
+    '("simple" "GST_AUDIO_CONVERT_NOISE_SHAPING_SIMPLE" "2")
+    '("medium" "GST_AUDIO_CONVERT_NOISE_SHAPING_MEDIUM" "3")
+    '("high" "GST_AUDIO_CONVERT_NOISE_SHAPING_HIGH" "4")
   )
 )
 
@@ -782,6 +769,17 @@
     '("event" "GST_GDP_DEPAY_STATE_EVENT" "4")
   )
 )
+;; Handwritten for GstGDPPay plugin (values taken from output of
+;; `gst-inspect gdppay' -- "version" property information)
+
+(define-enum-extended DPVersion
+  (in-module "Gst")
+  (c-name "GstDPVersion")
+  (values
+    '("0.2" "GST_GDP_VERSION_0_2" "1")
+    '("1.0" "GST_GDP_VERSION_1_0" "2")
+  )
+)
 
 ;; From /usr/src/jhbuild-sources/gst-plugins-base/gst/playback/gstfactorylists.h
 
@@ -1054,17 +1052,6 @@
   )
 )
 
-;; From /usr/src/jhbuild-sources/gst-plugins-base/ext/ogg/gstoggmux.h
-
-(define-enum-extended 
-  (in-module "")
-  (c-name "")
-  (values
-    '("control" "GST_OGG_PAD_STATE_CONTROL" "0")
-    '("data" "GST_OGG_PAD_STATE_DATA" "1")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-base/ext/pango/gsttextoverlay.h
 
 (define-enum-extended TextOverlayVAlign
@@ -1092,9 +1079,9 @@
   (c-name "GstTextOverlayWrapMode")
   (values
     '("none" "GST_TEXT_OVERLAY_WRAP_MODE_NONE" "-1")
-    '("word" "GST_TEXT_OVERLAY_WRAP_MODE_WORD" "#error")
-    '("char" "GST_TEXT_OVERLAY_WRAP_MODE_CHAR" "#error")
-    '("word-char" "GST_TEXT_OVERLAY_WRAP_MODE_WORD_CHAR" "#error")
+    '("word" "GST_TEXT_OVERLAY_WRAP_MODE_WORD" "0")
+    '("char" "GST_TEXT_OVERLAY_WRAP_MODE_CHAR" "1")
+    '("word-char" "GST_TEXT_OVERLAY_WRAP_MODE_WORD_CHAR" "2")
   )
 )
 
@@ -1102,21 +1089,35 @@
   (in-module "Gst")
   (c-name "GstTextOverlayLineAlign")
   (values
-    '("left" "GST_TEXT_OVERLAY_LINE_ALIGN_LEFT" "#error")
-    '("center" "GST_TEXT_OVERLAY_LINE_ALIGN_CENTER" "#error")
-    '("right" "GST_TEXT_OVERLAY_LINE_ALIGN_RIGHT" "#error")
+    '("left" "GST_TEXT_OVERLAY_LINE_ALIGN_LEFT" "0")
+    '("center" "GST_TEXT_OVERLAY_LINE_ALIGN_CENTER" "1")
+    '("right" "GST_TEXT_OVERLAY_LINE_ALIGN_RIGHT" "2")
   )
 )
 
 ;; From /usr/src/jhbuild-sources/gst-plugins-base/ext/theora/gsttheoraenc.h
 
-(define-enum-extended 
-  (in-module "")
-  (c-name "")
-  (values
-    '("none" "BORDER_NONE" "0")
-    '("black" "BORDER_BLACK" "1")
-    '("mirror" "BORDER_MIRROR" "2")
+(define-enum-extended TheoraEncBorderMode
+  (in-module "Gst")
+  (c-name "GstTheoraEncBorderMode")
+  (values
+    '("none" "GST_THEORA_ENC_BORDER_NONE" "0")
+    '("black" "GST_THEORA_ENC_BORDER_BLACK" "1")
+    '("mirror" "GST_THEORA_ENC_BORDER_MIRROR" "2")
   )
 )
 
+;; Handwritten for CdParanoiaSrc plugin (values taken from output of
+;; `gst-inspect cdparanoiasrc' -- "paranoia-mode" property information)
+(define-enum-extended GstCdParanoiaSrc
+  (in-module "Gst")
+  (c-name "GstCdParanoiaMode")
+  (values
+    '("disable" "GST_PARANOIA_MODE_DISABLE" "1<<0")
+    '("fragment" "GST_PARANOIA_MODE_FRAGMENT" "1<<1")
+    '("overlap" "GST_PARANOIA_MODE_OVERLAP" "1<<2")
+    '("scratch" "GST_PARANOIA_MODE_SCRATCH" "1<<3")
+    '("repair" "GST_PARANOIA_MODE_REPAIR" "1<<4")
+    '("full" "GST_PARANOIA_MODE_FULL" "0xff")
+  )
+)
