--- gstbase_enums.defs	2008-12-10 19:42:34.000000000 -0500
+++ gstbase_enums.defs.new	2008-12-10 19:42:09.000000000 -0500
@@ -707,10 +707,10 @@
   (in-module "Gst")
   (c-name "GstAudioConvertDithering")
   (values
-    '("none" "DITHER_NONE" "0")
-    '("rpdf" "DITHER_RPDF" "1")
-    '("tpdf" "DITHER_TPDF" "2")
-    '("tpdf-hf" "DITHER_TPDF_HF" "3")
+    '("none" "GST_AUDIO_CONVERT_DITHER_NONE" "0")
+    '("rpdf" "GST_AUDIO_CONVERT_DITHER_RPDF" "1")
+    '("tpdf" "GST_AUDIO_CONVERT_DITHER_TPDF" "2")
+    '("tpdf-hf" "GST_AUDIO_CONVERT_DITHER_TPDF_HF" "3")
   )
 )
 
@@ -718,11 +718,11 @@
   (in-module "Gst")
   (c-name "GstAudioConvertNoiseShaping")
   (values
-    '("none" "NOISE_SHAPING_NONE" "0")
-    '("error-feedback" "NOISE_SHAPING_ERROR_FEEDBACK" "1")
-    '("simple" "NOISE_SHAPING_SIMPLE" "2")
-    '("medium" "NOISE_SHAPING_MEDIUM" "3")
-    '("high" "NOISE_SHAPING_HIGH" "4")
+    '("none" "GST_AUDIO_CONVERT_NOISE_SHAPING_NONE" "0")
+    '("error-feedback" "GST_AUDIO_CONVERT_NOISE_SHAPING_ERROR_FEEDBACK" "1")
+    '("simple" "GST_AUDIO_CONVERT_NOISE_SHAPING_SIMPLE" "2")
+    '("medium" "GST_AUDIO_CONVERT_NOISE_SHAPING_MEDIUM" "3")
+    '("high" "GST_AUDIO_CONVERT_NOISE_SHAPING_HIGH" "4")
   )
 )
 
@@ -769,6 +769,16 @@
     '("event" "GST_GDP_DEPAY_STATE_EVENT" "4")
   )
 )
+;; Handwritten for GstGDPPay (gdppay plugin)
+
+(define-enum-extended DPVersion
+  (in-module "Gst")
+  (c-name "GstDPVersion")
+  (values
+    '("0.2" "GST_GDP_VERSION_0_2" "1")
+    '("1.0" "GST_GDP_VERSION_1_0" "2")
+  )
+)
 
 ;; From /usr/src/jhbuild-sources/gst-plugins-base/gst/playback/gstfactorylists.h
 
@@ -1041,17 +1051,6 @@
   )
 )
 
-;; From /usr/src/jhbuild-sources/gst-plugins-base/ext/ogg/gstoggmux.h
-
-(define-enum-extended 
-  (in-module "")
-  (c-name "")
-  (values
-    '("control" "GST_OGG_PAD_STATE_CONTROL" "0")
-    '("data" "GST_OGG_PAD_STATE_DATA" "1")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-base/ext/pango/gsttextoverlay.h
 
 (define-enum-extended TextOverlayVAlign
@@ -1079,9 +1078,9 @@
   (c-name "GstTextOverlayWrapMode")
   (values
     '("none" "GST_TEXT_OVERLAY_WRAP_MODE_NONE" "-1")
-    '("word" "GST_TEXT_OVERLAY_WRAP_MODE_WORD" "#error")
-    '("char" "GST_TEXT_OVERLAY_WRAP_MODE_CHAR" "#error")
-    '("word-char" "GST_TEXT_OVERLAY_WRAP_MODE_WORD_CHAR" "#error")
+    '("word" "GST_TEXT_OVERLAY_WRAP_MODE_WORD" "0")
+    '("char" "GST_TEXT_OVERLAY_WRAP_MODE_CHAR" "1")
+    '("word-char" "GST_TEXT_OVERLAY_WRAP_MODE_WORD_CHAR" "2")
   )
 )
 
@@ -1089,21 +1088,21 @@
   (in-module "Gst")
   (c-name "GstTextOverlayLineAlign")
   (values
-    '("left" "GST_TEXT_OVERLAY_LINE_ALIGN_LEFT" "#error")
-    '("center" "GST_TEXT_OVERLAY_LINE_ALIGN_CENTER" "#error")
-    '("right" "GST_TEXT_OVERLAY_LINE_ALIGN_RIGHT" "#error")
+    '("left" "GST_TEXT_OVERLAY_LINE_ALIGN_LEFT" "0")
+    '("center" "GST_TEXT_OVERLAY_LINE_ALIGN_CENTER" "1")
+    '("right" "GST_TEXT_OVERLAY_LINE_ALIGN_RIGHT" "2")
   )
 )
 
 ;; From /usr/src/jhbuild-sources/gst-plugins-base/ext/theora/gsttheoraenc.h
 
-(define-enum-extended 
-  (in-module "")
-  (c-name "")
-  (values
-    '("none" "BORDER_NONE" "0")
-    '("black" "BORDER_BLACK" "1")
-    '("mirror" "BORDER_MIRROR" "2")
+(define-enum-extended TheoraEncBorderMode
+  (in-module "Gst")
+  (c-name "GstTheoraEncBorderMode")
+  (values
+    '("none" "GST_THEORA_ENC_BORDER_NONE" "0")
+    '("black" "GST_THEORA_ENC_BORDER_BLACK" "1")
+    '("mirror" "GST_THEORA_ENC_BORDER_MIRROR" "2")
   )
 )
 
