--- gst_enums.defs.orig	2009-03-31 19:41:40.000000000 -0400
+++ gst_enums.defs	2009-04-02 16:52:21.000000000 -0400
@@ -25,15 +25,6 @@
 
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/flx/flx_color.h
 
-(define-enum-extended ColorSpaceType
-  (in-module "Flx")
-  (c-name "FlxColorSpaceType")
-  (values
-    '("rgb8" "FLX_COLORSPACE_RGB8" "0")
-    '("rgb32" "FLX_COLORSPACE_RGB32" "1")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/flx/gstflxdec.h
 
 (define-enum-extended FlxDecState
@@ -47,30 +38,8 @@
 
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/goom/goom_config_param.h
 
-(define-enum-extended Type
-  (in-module "Param")
-  (c-name "ParamType")
-  (values
-    '("intval" "PARAM_INTVAL" "0")
-    '("floatval" "PARAM_FLOATVAL" "1")
-    '("boolval" "PARAM_BOOLVAL" "2")
-    '("strval" "PARAM_STRVAL" "3")
-    '("listval" "PARAM_LISTVAL" "4")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/id3demux/id3tags.h
 
-(define-enum-extended D3TagsResult
-  (in-module "I")
-  (c-name "ID3TagsResult")
-  (values
-    '("more-data" "ID3TAGS_MORE_DATA" "0")
-    '("read-tag" "ID3TAGS_READ_TAG" "1")
-    '("broken-tag" "ID3TAGS_BROKEN_TAG" "2")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/matroska/matroska-demux.h
 
 (define-enum-extended MatroskaDemuxState
@@ -121,13 +90,6 @@
   )
 )
 
-(define-enum-extended MatroskaVideoTrackFlags
-  (in-module "Gst")
-  (c-name "GstMatroskaVideoTrackFlags")
-  (values
-  )
-)
-
 (define-flags-extended MatroskaTrackEncodingScope
   (in-module "Gst")
   (c-name "GstMatroskaTrackEncodingScope")
@@ -163,56 +125,14 @@
 
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/oldcore/gstaggregator.h
 
-(define-enum-extended AggregatorSchedType
-  (in-module "Gst")
-  (c-name "GstAggregatorSchedType")
-  (values
-    '("loop" "AGGREGATOR_LOOP" "1")
-    '("loop-select" "AGGREGATOR_LOOP_SELECT" "2")
-    '("chain" "AGGREGATOR_CHAIN" "3")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/oldcore/gstmultifilesrc.h
 
-(define-enum-extended MultiFileSrcFlags
-  (in-module "Gst")
-  (c-name "GstMultiFileSrcFlags")
-  (values
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/oldcore/gstpipefilter.h
 
-(define-enum-extended PipeFilterFlags
-  (in-module "Gst")
-  (c-name "GstPipeFilterFlags")
-  (values
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/oldcore/gstshaper.h
 
-(define-enum-extended ShaperPolicyType
-  (in-module "Gst")
-  (c-name "GstShaperPolicyType")
-  (values
-    '("timestamps" "SHAPER_POLICY_TIMESTAMPS" "1")
-    '("buffersize" "SHAPER_POLICY_BUFFERSIZE" "2")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/qtdemux/qtdemux_types.h
 
-(define-flags-extended Flags
-  (in-module "Qt")
-  (c-name "QtFlags")
-  (values
-    '("none" "QT_FLAG_NONE" "#error")
-    '("container" "QT_FLAG_CONTAINER" "1 << 0")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/rtp/gstrtpamrdepay.h
 
 (define-enum-extended RtpAMRDepayMode
@@ -274,51 +194,12 @@
 
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/rtp/gstrtpilbcdepay.h
 
-(define-enum-extended LBCMode
-  (in-module "Gsti")
-  (c-name "GstiLBCMode")
-  (values
-    '("20" "GST_ILBC_MODE_20" "20")
-    '("30" "GST_ILBC_MODE_30" "30")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/rtsp/gstrtspsrc.h
 
-(define-enum-extended RTSPNatMethod
-  (in-module "Gst")
-  (c-name "GstRTSPNatMethod")
-  (values
-    '("none" "GST_RTSP_NAT_NONE" "0")
-    '("dummy" "GST_RTSP_NAT_DUMMY" "1")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/udp/gstudp.h
 
-(define-enum-extended UDPControl
-  (in-module "Gst")
-  (c-name "GstUDPControl")
-  (values
-    '("zero" "CONTROL_ZERO" "0")
-    '("none" "CONTROL_NONE" "1")
-    '("udp" "CONTROL_UDP" "2")
-    '("tcp" "CONTROL_TCP" "3")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/videocrop/gstvideocrop.h
 
-(define-enum-extended CropPixelFormat
-  (in-module "Video")
-  (c-name "VideoCropPixelFormat")
-  (values
-    '("packed-simple" "VIDEO_CROP_PIXEL_FORMAT_PACKED_SIMPLE" "0")
-    '("packed-complex" "VIDEO_CROP_PIXEL_FORMAT_PACKED_COMPLEX" "1")
-    '("planar" "VIDEO_CROP_PIXEL_FORMAT_PLANAR" "2")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/videofilter/gstvideoflip.h
 
 (define-enum-extended VideoFlipMethod
@@ -338,110 +219,40 @@
 
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/videomixer/videomixer.h
 
-(define-enum-extended 
-  (in-module "")
-  (c-name "")
-  (values
-    '("checker" "VIDEO_MIXER_BACKGROUND_CHECKER" "0")
-    '("black" "VIDEO_MIXER_BACKGROUND_BLACK" "1")
-    '("white" "VIDEO_MIXER_BACKGROUND_WHITE" "2")
-  )
-)
-
-;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/wavenc/riff.h
-
-(define-enum-extended RiffReturn
+(define-enum-extended VideoMixerBackground
   (in-module "Gst")
-  (c-name "GstRiffReturn")
+  (c-name "GstVideoMixerBackground")
   (values
-    '("ok" "GST_RIFF_OK" "0")
-    '("enotriff" "GST_RIFF_ENOTRIFF" "-1")
-    '("einval" "GST_RIFF_EINVAL" "-2")
-    '("enomem" "GST_RIFF_ENOMEM" "-3")
+    '("checker" "GST_VIDEO_MIXER_BACKGROUND_CHECKER" "0")
+    '("black" "GST_VIDEO_MIXER_BACKGROUND_BLACK" "1")
+    '("white" "GST_VIDEO_MIXER_BACKGROUND_WHITE" "2")
   )
 )
 
-;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/wavparse/gstwavparse.h
+;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/wavenc/riff.h
 
-(define-enum-extended WavParseState
-  (in-module "Gst")
-  (c-name "GstWavParseState")
-  (values
-    '("start" "GST_WAVPARSE_START" "0")
-    '("header" "GST_WAVPARSE_HEADER" "1")
-    '("data" "GST_WAVPARSE_DATA" "2")
-  )
-)
+;; From /usr/src/jhbuild-sources/gst-plugins-good/gst/wavparse/gstwavparse.h
 
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/ext/annodex/gstcmmldec.h
 
-(define-enum-extended 
-  (in-module "")
-  (c-name "")
-  (values
-    '("ident-header" "GST_CMML_PACKET_IDENT_HEADER" "1")
-    '("first-header" "GST_CMML_PACKET_FIRST_HEADER" "2")
-    '("second-header" "GST_CMML_PACKET_SECOND_HEADER" "3")
-    '("clip" "GST_CMML_PACKET_CLIP" "4")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/ext/annodex/gstcmmlparser.h
 
-(define-enum-extended 
-  (in-module "")
-  (c-name "")
-  (values
-    '("e" "GST_CMML_PARSER_DECODE" "6")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/ext/cairo/gsttextoverlay.h
 
-(define-enum-extended 
-  (in-module "")
-  (c-name "")
-  (values
-    '("bottom" "GST_CAIRO_TEXT_OVERLAY_VALIGN_BOTTOM" "1")
-    '("top" "GST_CAIRO_TEXT_OVERLAY_VALIGN_TOP" "2")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/ext/esd/esdmon.h
 
-(define-enum-extended EsdSrcFlags
-  (in-module "Gst")
-  (c-name "GstEsdSrcFlags")
-  (values
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/ext/flac/gstflactag.h
 
-(define-enum-extended 
-  (in-module "")
-  (c-name "")
-  (values
-    '("init" "GST_FLAC_TAG_STATE_INIT" "0")
-    '("metadata-blocks" "GST_FLAC_TAG_STATE_METADATA_BLOCKS" "1")
-    '("metadata-next-block" "GST_FLAC_TAG_STATE_METADATA_NEXT_BLOCK" "2")
-    '("writing-metadata-block" "GST_FLAC_TAG_STATE_WRITING_METADATA_BLOCK" "3")
-    '("vc-metadata-block" "GST_FLAC_TAG_STATE_VC_METADATA_BLOCK" "4")
-    '("add-vorbis-comment" "GST_FLAC_TAG_STATE_ADD_VORBIS_COMMENT" "5")
-    '("audio-data" "GST_FLAC_TAG_STATE_AUDIO_DATA" "6")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/ext/gconf/gstgconf.h
 
 (define-enum-extended GConfProfile
   (in-module "Gst")
   (c-name "GstGConfProfile")
   (values
-    '("sounds" "GCONF_PROFILE_SOUNDS" "0")
-    '("music" "GCONF_PROFILE_MUSIC" "1")
-    '("chat" "GCONF_PROFILE_CHAT" "2")
-    '("none" "GCONF_PROFILE_NONE" "3")
+    '("sounds" "GST_GCONF_PROFILE_SOUNDS" "0")
+    '("music" "GST_GCONF_PROFILE_MUSIC" "1")
+    '("chat" "GST_GCONF_PROFILE_CHAT" "2")
+    '("none" "GST_GCONF_PROFILE_NONE" "3")
   )
 )
 
@@ -451,69 +262,19 @@
   (in-module "Gst")
   (c-name "GstPixbufScaleMethod")
   (values
-    '("nearest" "GST_PIXBUFSCALE_NEAREST" "0")
-    '("tiles" "GST_PIXBUFSCALE_TILES" "1")
-    '("bilinear" "GST_PIXBUFSCALE_BILINEAR" "2")
-    '("hyper" "GST_PIXBUFSCALE_HYPER" "3")
+    '("nearest" "GST_PIXBUF_SCALE_METHOD_NEAREST" "0")
+    '("tiles" "GST_PIXBUF_SCALE_METHOD_TILES" "1")
+    '("bilinear" "GST_PIXBUF_SCALE_METHOD_BILINEAR" "2")
+    '("hyper" "GST_PIXBUF_SCALE_METHOD_HYPER" "3")
   )
 )
 
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/ext/hal/hal.h
 
-(define-enum-extended HalDeviceType
-  (in-module "Gst")
-  (c-name "GstHalDeviceType")
-  (values
-    '("audiosink" "GST_HAL_AUDIOSINK" "0")
-    '("audiosrc" "GST_HAL_AUDIOSRC" "1")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/ext/jpeg/smokecodec.h
 
-(define-enum-extended CodecResult
-  (in-module "Smoke")
-  (c-name "SmokeCodecResult")
-  (values
-    '("wrongversion" "SMOKECODEC_WRONGVERSION" "-5")
-    '("wrongsize" "SMOKECODEC_WRONGSIZE" "-4")
-    '("error" "SMOKECODEC_ERROR" "-3")
-    '("nomem" "SMOKECODEC_NOMEM" "-2")
-    '("nullptr" "SMOKECODEC_NULLPTR" "-1")
-    '("ok" "SMOKECODEC_OK" "0")
-  )
-)
-
-(define-flags-extended CodecFlags
-  (in-module "Smoke")
-  (c-name "SmokeCodecFlags")
-  (values
-    '("keyframe" "SMOKECODEC_KEYFRAME" "1<<0")
-    '("motion-vectors" "SMOKECODEC_MOTION_VECTORS" "1<<1")
-  )
-)
-
-(define-flags-extended PacketType
-  (in-module "Smoke")
-  (c-name "SmokePacketType")
-  (values
-    '("id" "SMOKECODEC_TYPE_ID" "0x80")
-    '("comment" "SMOKECODEC_TYPE_COMMENT" "0x81")
-    '("extra" "SMOKECODEC_TYPE_EXTRA" "0x83")
-    '("data" "SMOKECODEC_TYPE_DATA" "0x40")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/ext/ladspa/gstsignalprocessor.h
 
-(define-flags-extended SignalProcessorClassFlags
-  (in-module "Gst")
-  (c-name "GstSignalProcessorClassFlags")
-  (values
-    '("e" "GST_SIGNAL_PROCESSOR_CLASS_FLAG_CAN_PROCESS_IN_PLACE" "1<<0")
-  )
-)
-
 (define-enum-extended SignalProcessorState
   (in-module "Gst")
   (c-name "GstSignalProcessorState")
@@ -526,25 +287,15 @@
 
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/ext/pulse/pulsemixerctrl.h
 
-(define-enum-extended PulseMixerType
-  (in-module "Gst")
-  (c-name "GstPulseMixerType")
-  (values
-    '("unknown" "GST_PULSEMIXER_UNKNOWN" "0")
-    '("sink" "GST_PULSEMIXER_SINK" "1")
-    '("source" "GST_PULSEMIXER_SOURCE" "2")
-  )
-)
-
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/ext/shout2/gstshout2.h
 
 (define-enum-extended Shout2SendProtocol
   (in-module "Gst")
   (c-name "GstShout2SendProtocol")
   (values
-    '("xaudiocast" "SHOUT2SEND_PROTOCOL_XAUDIOCAST" "1")
-    '("icy" "SHOUT2SEND_PROTOCOL_ICY" "2")
-    '("http" "SHOUT2SEND_PROTOCOL_HTTP" "3")
+    '("xaudiocast" "GST_SHOUT2SEND_PROTOCOL_XAUDIOCAST" "1")
+    '("icy" "GST_SHOUT2SEND_PROTOCOL_ICY" "2")
+    '("http" "GST_SHOUT2SEND_PROTOCOL_HTTP" "3")
   )
 )
 
@@ -562,15 +313,3 @@
 )
 
 ;; From /usr/src/jhbuild-sources/gst-plugins-good/ext/speex/gstspeexenc.h
-
-(define-enum-extended SpeexMode
-  (in-module "Gst")
-  (c-name "GstSpeexMode")
-  (values
-    '("auto" "GST_SPEEX_ENC_MODE_AUTO" "0")
-    '("uwb" "GST_SPEEX_ENC_MODE_UWB" "1")
-    '("wb" "GST_SPEEX_ENC_MODE_WB" "2")
-    '("nb" "GST_SPEEX_ENC_MODE_NB" "3")
-  )
-)
-
